lib/multi-sse.js

```diff
@@ -0,0 +1,128 @@
+var _ = require('lodash');
+var mongojs = require('mongojs');
+var inflect = require('i')();
+var tinySSE = require('tiny-sse');
+var SSE = require('./sse');
+var Promise = require('bluebird');
+var hl = require('highland');
+var JSONAPI_Error = require('./jsonapi-error');
+
+var MultiSSE = function() {
+};
+
+MultiSSE.prototype.init = function(harvesterApp) {
+    this.options = harvesterApp.options;
+    this.db = mongojs(this.options.oplogConnectionString);
+    this.coll = this.db.collection('oplog.rs');
+    this.harvesterApp = harvesterApp;
+    harvesterApp.router.get(this.options.baseUrl + '/changes/stream', this.checkValidRequestMiddleware.bind(this), tinySSE.head(), tinySSE.ticker({seconds: 3}), this.handler.bind(this));
+};
+
+MultiSSE.prototype.checkValidRequestMiddleware = function (req, res, next) {
+    routeNames = req.query.resources ? req.query.resources.split(',') : [];
+
+    if (routeNames.length === 0) {
+        throw new JSONAPI_Error({
+            status: 400,
+            title: 'Requested changes on missing resource',
+            detail: 'You have not specified any resources, please do so by providing "resource?foo,bar" as query'
+        });
+    }
+
+    if (!this.allResourcesExist(routeNames)) {
+        throw new JSONAPI_Error({
+            status: 400,
+            title: 'Requested changes on missing resource',
+            detail: 'The follow resources don\'t exist ' +  this.getMissingResources(routeNames).join(',')
+        });
+    }
+
+    if (req.headers['last-event-id']) {
+        var tsSplit = _.map(req.headers['last-event-id'].split('_'), function (item) {
+            return parseInt(item, 10);
+        });
+
+        isValidTS = _.all(tsSplit, function(ts) {
+            return !isNaN(ts);
+        });
+
+        if(!isValidTS) {
+            throw new JSONAPI_Error({
+                status: 400,
+                title: 'Invalid Timestamp',
+                detail: 'Could not parse the time stamp provided'
+            });
+        }
+    }
+
+    next();
+};
+
+MultiSSE.prototype.handler = function (req, res, next) {
+    var me = this;
```

Comment:
not sure if me as this is a typical convention. I've seen _this & that, but not me. This is partially me wanting to keep things similar, and also me being curious about whether this is a new convention.